tests/DCPS/Compiler/optional/main.cpp

```diff
@@ -1,16 +1,106 @@
+#include "optionalC.h"
 #include "optionalTypeSupportImpl.h"
+#include "tests/Utils/DataView.h"
 
+#include "gtest/gtest.h"
 #include <dds/DCPS/Serializer.h>
 #include <dds/DCPS/TypeSupportImpl.h>
 #include <dds/DCPS/FilterEvaluator.h>
 
+#include <vector>
 #include <ace/ACE.h>
 #include <ace/Log_Msg.h>
+#include <gtest/gtest.h>
+#include <tests/Utils/DataView.h>
 
-#include <cstdlib>
+using namespace OpenDDS::DCPS;
+using namespace OpenDDS::XTypes;
 
+const Encoding xcdr1(Encoding::KIND_XCDR1, ENDIAN_BIG);
+const Encoding xcdr2(Encoding::KIND_XCDR2, ENDIAN_BIG);
+const Encoding xcdr2_le(Encoding::KIND_XCDR2, ENDIAN_LITTLE);
 
-int ACE_TMAIN(int, ACE_TCHAR**)
+template<typename Type>
+void serializer_test(const OpenDDS::DCPS::Encoding& encoding, Type value, const DataView& expected_cdr) {
```

Comment:
I'm honestly not sure about serialization tests being here. The xcdr test would probably be a better place for them as it can then compare generated serialization vs dynamic serialization (even if we don't have that working yet).

Reply:
I can move the tests, although for now since I've been using cmake to build and test I'll move this near the end of the PR, unless it's fine to add them to the cmake definitions ahead of your other [PR](https://github.com/OpenDDS/OpenDDS/pull/4374)

Reply:
what would be the best way to add these tests to the xcdr test. The xcdr tests don't use C++11 I think, I noticed other tests being split between C++11 and C++03, should it be done the same way as the others or is there something else to be done here?

Reply:
@iguessthislldo @sonndinh Thoughts?